lib/core/store.ts

```diff
@@ -132,4 +145,61 @@ export interface IPromiseStore {
     tags?: Record<string, string>,
     limit?: number,
   ): AsyncGenerator<DurablePromise[], void>;
+
+  /**
+   * Creates a new schedule.
+   *
+   * @param id Unique identifier for the schedule.
+   * @param ikey Idempotency key associated with the create operation.
+   * @param description Description of the schedule.
+   * @param cron CRON expression defining the schedule's execution time.
+   * @param tags Key-value pairs associated with the schedule.
+   * @param promiseId Unique identifier for the associated promise.
+   * @param promiseTimeout Timeout for the associated promise in milliseconds.
+   * @param promiseHeaders Headers associated with the promise data.
+   * @param promiseData Encoded data for the promise of type string.
+   * @param promiseTags Key-value pairs associated with the promise.
+   * @returns A Promise resolving to the created schedule.
+   */
+  createSchedule(
```

Comment:
Let's move the schedule functions to a separate interface called `IScheduleStore`. This should help with the asymmetry.